Neos.Flow/Classes/Persistence/Doctrine/EntityManagerConfiguration.php

```diff
@@ -168,16 +175,19 @@ protected function applyDqlSettingsToConfiguration(array $configuredSettings, Co
      */
     protected function applyCacheConfiguration(Configuration $config): void
     {
-        $cache = new CacheAdapter();
-        // must use ObjectManager in compile phase...
-        $cache->setCache($this->objectManager->get(CacheManager::class)->getCache('Flow_Persistence_Doctrine'));
-        $config->setMetadataCacheImpl($cache);
-        $config->setQueryCacheImpl($cache);
-
-        $resultCache = new CacheAdapter();
-        // must use ObjectManager in compile phase...
-        $resultCache->setCache($this->objectManager->get(CacheManager::class)->getCache('Flow_Persistence_Doctrine_Results'));
-        $config->setResultCacheImpl($resultCache);
+        // Here we do not use the wrapper as below as the metadata cannot change at runtime anyway.
+        $cache = $this->objectManager->get(CacheManager::class)->getCacheItemPool('Flow_Persistence_Doctrine_Metadata');
+        $config->setMetadataCache($cache);
+
+        /**
+         * FIXME:
+         * We shouldn't need this wrapper adding the security hash as {@see SqlFilter::addFilterConstraint} does that already,
+         * and the parameters there are hashed into the query cache key in doctrines Query class.
+         * But tests fail if it doesn't happen
+         */
+        $config->setQueryCache($this->getSecurityHashAwareCacheItemPool('Flow_Persistence_Doctrine'));
+
+        $config->setResultCache($this->getSecurityHashAwareCacheItemPool('Flow_Persistence_Doctrine_Results'));
```

Comment:
i dont understand the implications here fully ... we have a new cache `Flow_Persistence_Doctrine_Metadata`? but why?

Reply:
Because before two different things went into one cache, which technically is bad because we don't contorl it. if doctrine flushes the query parser cache the metadata would be flusghed as well -> not useful. Also one needs the security hash, the other definitely not.

Reply:
Should we rename `Flow_Persistence_Doctrine` to `Flow_Persistence_Doctrine_Query` then to better distinct them?

Reply:
I dind't want to break existing Cache configurations 🤷

Reply:
Like this you _might_ have the metadata cache in files instead of <chosen cache backend> but it would work, if we rename Flow_Persistence_Doctrine your cache config will throw

Reply:
Oke and regarding 

> But tests fail if it doesn't happen

then the tests are wrong i guess? Or is it too much to adjust?

Reply:
Well I only see test failures, but these tests should be correct, emphasis on should. I think this is something we need to figure out with some people deeper into the security topic. I think there could definitely be some simplification here, but this is how it was before so at least nothing broken for now.

Reply:
i decided to get this change in to keep it as is and discuss about it further afterwards

Reply:
okay sounds fine i +1